device/core/src/blob_upload/file_upload_api.ts

```diff
@@ -37,107 +37,110 @@ export class FileUploadApi implements FileUploadInterface {
+                const body = JSON.stringify(uploadResult);
+
+                /*Codes_SRS_NODE_FILE_UPLOAD_ENDPOINT_16_014: [The `POST` HTTP request shall have the following headers:
+                ```
+                Host: <hostname>,
+                Authorization: <sharedAccessSignature>,
+                'User-Agent': <version>,
                 'Content-Type': 'application/json; charset=utf-8',
-                'Content-Length': body.length,
-                'iothub-name': deviceCredentials.host.split('.')[0]
-            };
-
-            let x509Opts = null;
-            if (typeof deviceCredentials.sharedAccessSignature === 'string') {
-                headers.Authorization = deviceCredentials.sharedAccessSignature;
-            } else {
-                x509Opts = deviceCredentials.x509;
-            }
-
-            const req = this.http.buildRequest('POST', path, headers, deviceCredentials.host, x509Opts, (err) => {
-                if (err) {
-                    /*Codes_SRS_NODE_FILE_UPLOAD_ENDPOINT_16_016: [** `notifyUploadComplete` shall call the `done` callback with an `Error` object if the request fails.]*/
-                    done(err);
+                'Content-Length': <content length>,
+                'iothub-name': <hub name>
+                ```]*/
+                let headers: any = {
+                    Host: deviceCredentials.host,
+                    'User-Agent': packageJson.name + '/' + packageJson.version,
+                    'Content-Type': 'application/json; charset=utf-8',
+                    'Content-Length': body.length,
+                    'iothub-name': deviceCredentials.host.split('.')[0]
+                };
+
+                let x509Opts = null;
+                if (typeof deviceCredentials.sharedAccessSignature === 'string') {
+                    headers.Authorization = deviceCredentials.sharedAccessSignature;
                 } else {
-                    /*Codes_SRS_NODE_FILE_UPLOAD_ENDPOINT_16_017: [** `notifyUploadComplete` shall call the `done` callback with no parameters if the request succeeds.]*/
-                    done();
+                    x509Opts = deviceCredentials.x509;
                 }
+
+                const req = this.http.buildRequest('POST', path, headers, deviceCredentials.host, x509Opts, (err) => {
+                    if (err) {
+                        /*Codes_SRS_NODE_FILE_UPLOAD_ENDPOINT_16_016: [** `notifyUploadComplete` shall call the `done` callback with an `Error` object if the request fails.]*/
+                        _callback(err);
+                    } else {
+                        /*Codes_SRS_NODE_FILE_UPLOAD_ENDPOINT_16_017: [** `notifyUploadComplete` shall call the `done` callback with no parameters if the request succeeds.]*/
+                        _callback();
```

Comment:
>_callback() [](start = 24, length = 11)

This is NOT a problem you introduced, but the original version here had a ? on the done.  The original code here SHOULD  have tested if done was falsy.

If it had been done, this would have made this change difficult like what is currently happening in the provisioningserviceclient _delete method.

As it is, I want to discuss this with Pierre.